src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -490,3 +526,201 @@ def execute_hdi_script_action(cmd, client, resource_group_name, cluster_name,
+        autoscale_configuration.recurrence = AutoscaleRecurrence(
+            schedule=[condition]
+        )
+
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def update_autoscale_condition(cmd, client, resource_group_name, cluster_name, index, days=None, time=None, count=None):
+    autoscale_configuration = None
+    cluster = client.get(resource_group_name, cluster_name)
+    for role in cluster.properties.compute_profile.roles:
+        if role.name == "workernode":
+            autoscale_configuration = role.autoscale_configuration
+            break
+    if not autoscale_configuration:
+        raise CLIError('The cluster has not enabled autoscale.')
+
+    if not autoscale_configuration.recurrence:
+        raise CLIError('The cluster does not have any schedule conditions to update.')
+
+    index = int(index)
+    if index >= len(autoscale_configuration.recurrence.schedule):
+        raise CLIError(
+            'This cluster only has {} condition(s). Please specify a correct index which starts with 0.'.format(
+                len(autoscale_configuration.recurrence.schedule)))
+
+    # try to update schedule condition
+    if days:
+        autoscale_configuration.recurrence.schedule[index].days = days
+    if time:
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.time = time
+    if count:
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.min_instance_count = count
+        autoscale_configuration.recurrence.schedule[index].time_and_capacity.max_instance_count = count
+
+    return client.update_auto_scale_configuration(resource_group_name, cluster_name, autoscale_configuration)
+
+
+def delete_autoscale_condition(cmd, client, resource_group_name, cluster_name, index):
+    autoscale_configuration = None
+    cluster = client.get(resource_group_name, cluster_name)
+    for role in cluster.properties.compute_profile.roles:
+        if role.name == "workernode":
+            autoscale_configuration = role.autoscale_configuration
+            break
+    if not autoscale_configuration:
+        raise CLIError('The cluster has not enabled autoscale.')
+
+    if not autoscale_configuration.recurrence:
+        raise CLIError('The cluster does not have any schedule conditions to delete.')
+
+    if len(index) >= len(autoscale_configuration.recurrence.schedule):
+        raise CLIError('Deleting all conditions is not allowed.'
+                       'If you want to disable autoscale please use `az hdinsight autoscale delete`')
+
+    for i in index:
+        i = int(i)
+        if i >= len(autoscale_configuration.recurrence.schedule):
+            raise CLIError(
+                'This cluster only has {} conditions. Please specify a correct index which starts with 0.'.format(
+                    len(autoscale_configuration.recurrence.schedule)))
+        autoscale_configuration.recurrence.schedule.pop(i)
```

Comment:
let's say we have two conditions at the beginning. Can `az synapse autoscale delete ... --index 0 1` successfully remove these two rules?

Reply:
No it can't. It will raise CLI Error due to
```
 if len(index) >= len(autoscale_configuration.recurrence.schedule):
        raise CLIError('Deleting all conditions is not allowed.'
                       'If you want to disable autoscale please use `az hdinsight autoscale delete`')
```

---
In reply to: [466356086](https://github.com/Azure/azure-cli/pull/14692#discussion_r466356086) [](ancestors = 466356086)

Reply:
Oh, sorry. Consider the scenario there are 3 conditions and the user want to remove the first two. Have you tested this code logic? I suppose there might be some unexpected results...

Reply:
I think that I got your point. I should store the len of autoscale_configuration.recurrence.schedule. I will fix it.

---
In reply to: [466379128](https://github.com/Azure/azure-cli/pull/14692#discussion_r466379128) [](ancestors = 466379128)

Reply:
You find the bug in monitor too.

---
In reply to: [466379128](https://github.com/Azure/azure-cli/pull/14692#discussion_r466379128) [](ancestors = 466379128)

Reply:
fixed.

---
In reply to: [466851735](https://github.com/Azure/azure-cli/pull/14692#discussion_r466851735) [](ancestors = 466851735,466379128)